src/controllers/blocks/BlocksController.ts

```diff
@@ -233,4 +236,72 @@ export default class BlocksController extends AbstractController<BlocksService>
 			await this.service.fetchBlockHeader(hash)
 		);
 	};
+
+	/**
+	 * Return a collection of blocks, given a range.
+	 *
+	 * @param req Express Request
+	 * @param res Express Response
+	 */
+	private getBlocks: RequestHandler<
+		unknown,
+		unknown,
+		unknown,
+		IRangeQueryParam
+	> = async (
+		{ query: { range, eventDocs, extrinsicDocs } },
+		res
+	): Promise<void> => {
+		if (!range) throw new BadRequest('range query parameter must be inputted.');
+
+		// We set a max range to 500 blocks.
+		const rangeOfNums = this.parseRangeOfNumbersOrThrow(range, 500);
+		const rangeOfNumsToHash = await Promise.all(
+			rangeOfNums.map(async (n) => await this.getHashForBlock(n.toString()))
```

Comment:
Should this be queued at all too or happy to do all of the number -> hash requests at once, as opposed to block details which are queued to do 4 at a time?

Reply:
I almost wonder whether you could combine a bunch of steps into something like:

```
const pQueue = new PromiseQueue(4);
const blocksPromise: Promise<Unknown>[] = [];

for (let i = 0; i < rangeOfNums.length; i++) {
    let result = pQueue.push(async () => {
        let hash = await this.getHashForBlock(n.toString());
	const historicApi = await this.api.at(hash);
	return await this.service.fetchBlock(
		hash,
		historicApi,
		options
	);
    });
    blocksPromise.push(result);
}
```

That way there would be no intermediate array for tasks and no intermediate array for block hashes, and also everything is queued, rather than hashes being all obtained at once. WDYT?

Reply:
Yea something like this would be cleaner. Having it all in one iteration is way more sound IMO. I agree. Will update it right now to work, and run some quick benchmarks.

Reply:
Thanks for the suggestion.